homeassistant/components/zha/radio_manager.py

```diff
@@ -324,9 +335,14 @@ async def async_initiate_migration(self, data: dict[str, Any]) -> bool:
         # Temporarily connect to the old radio to read its settings
         old_radio_mgr = ZhaRadioManager()
         old_radio_mgr.hass = self._hass
-        old_radio_mgr.radio_type = old_radio_type
-        old_radio_mgr.device_path = old_device_settings[CONF_DEVICE_PATH]
-        old_radio_mgr.device_settings = old_device_settings
+        old_radio_mgr.device_path = old_device_path
+        if "hw" in migration_data["old_discovery_info"]:
+            old_radio_mgr.radio_type = old_radio_type
+            old_radio_mgr.device_settings = old_device_settings
+        else:
+            if not await old_radio_mgr.detect_radio_type():
+                # Could not talk to the USB port, do nothing
+                return False
```

Comment:
I'm not sure if this is correct:
- Is the detection needed?
- If detection fails, should the config entry not be updated, even if it's configured to use the USB connected radio?

Reply:
The current flow is to connect to the old radio, read its settings, then start the flashing. Shouldn't this be independent of the migration data?

Reply:
I think this has been resolved @puddly ?